sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/PodSpecsCannotShrink.java

```diff
@@ -32,16 +32,24 @@
         Collection<ConfigValidationError> errors = new ArrayList<>();
         for (PodSpec oldPod : oldConfig.get().getPods()) {
             PodSpec newPod = newPods.get(oldPod.getType());
-            if (newPod == null) {
+            if (newPod == null && !oldPod.getAllowDecommission()) {
```

Comment:
It seems as if we could just `continue` on `oldPod.getAllowDecommission()` instead of adding this to each of the conditionals?

Reply:
That would indeed be better -- the catch is that it's checking oldPod only when newPod is null. Otherwise it checks newPod. However these clauses are indeed pretty dense now. Broke them up a bit, adding a continue in the null case. It should hopefully be a bit less confusing